modules/tinymce/src/themes/silver/main/ts/ui/window/SilverDialogEvents.ts

```diff
@@ -116,19 +92,23 @@ const initDialog = <T>(getInstanceApi: () => Types.Dialog.DialogInstanceApi<T>,
 
     fireApiEvent<FormActionEvent>(formActionEvent, (api, spec, event, component) => {
       const focusIn = () => Keying.focusIn(component);
+      const isDisabled = (focused: Element<HTMLElement>) => Attr.has(focused, 'disabled') || Attr.getOpt(focused, 'aria-disabled').exists((val) => val === 'true');
       const current = Focus.active();
 
       spec.onAction(api, { name: event.name(), value: event.value() });
 
-      Focus.active().fold(() => {
-        focusIn();
-      }, (focused) => {
+      Focus.active().fold(focusIn, (focused) => {
         // We need to check if the focused element is disabled because apparently firefox likes to leave focus on disabled elements.
-        if (!Compare.contains(component.element(), focused) || Attr.has(focused, 'disabled')) {
+        if (isDisabled(focused)) {
           focusIn();
-          // And we need the below check for IE, which likes to leave focus on the parent of disabled elements
-        } else if (Compare.contains(focused, current.getOrNull()) && Attr.has(current.getOrDie(), 'disabled')) {
+        // And we need the below check for IE, which likes to leave focus on the parent of disabled elements
+        } else if (current.exists((cur) => Compare.contains(focused, cur) && isDisabled(cur))) {
```

Comment:
combine this branch and the if? though readability...

Reply:
Yeah, I didn't do that specifically because the comments/logic becomes more confusing.